src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -1058,6 +1058,12 @@
     <Field Name="LabelExpressionOpt" Type="BoundLabel" Null="allow"/>
   </Node>
 
+  <!-- only used by codegen -->
+  <Node Name="BoundGotoExpression" Base="BoundExpression">
```

Comment:
One of the major points for the expression/statement separation is that expressions do not interfere with control flow. The node breaks the important invariant and is added to the tree very early.

We need a different solution. Introducing goto expression is too big of a change, architecturally, to handle this issue.

Reply:
This is a codegen-only node and is lowered similar to throw-expressions (emits a default to adjust invariants)

Also https://github.com/dotnet/csharplang/issues/176 would actually *rely* on the possibility that we can jump out of expressions.. I saw a comment from AlekseyTs worth to quote here (https://github.com/dotnet/roslyn/pull/24263#discussion_r162220656): "Short-circuiting binary operators imply a control flow." wouldn't that apply here?